test/utils/scheduler/event-builder.ts

```diff
@@ -0,0 +1,210 @@
+import {
+  CalendarResourceId,
+  RecurringEventPresetKey,
+  RecurringEventRecurrenceRule,
+} from '@mui/x-scheduler-headless/models';
+import {
+  CalendarEvent,
+  CalendarEventId,
+  CalendarEventOccurrence,
+} from '@mui/x-scheduler-headless/models/event';
+import { getWeekDayCode } from '@mui/x-scheduler-headless/utils/recurring-event-utils';
+import { Adapter } from '@mui/x-scheduler-headless/use-adapter';
+import { adapter as defaultAdapter } from './adapters';
+
+const DEFAULT_TESTING_VISIBLE_DATE_STR = '2025-07-03T00:00:00Z';
+export const DEFAULT_TESTING_VISIBLE_DATE = defaultAdapter.date(DEFAULT_TESTING_VISIBLE_DATE_STR);
+
+/**
+ * Minimal event builder for tests.
+ *
+ * Scope:
+ * - Only sets fields and time ranges.
+ * - Uses the provided (or default) adapter for all date ops.
+ * - Returns a CalendarEventOccurrence for reducing test boilerplate.
+ */
+export class EventBuilder {
+  /** Internal draft; only partial until `build()` time. */
+  protected event: Partial<CalendarEvent> = { id: crypto.randomUUID() };
```

Comment:
Is there a reason not to generate the default title here as well and use `event: CalendarEvent`?
It would be safer if tomorrow we have other required properties.